Nodejs/Product/Analysis/Analysis/AnalysisLimits.cs

```diff
@@ -34,6 +34,17 @@ internal class AnalysisLimits {
             MaxObjectLiteralProperties = 50;
             MaxObjectKeysTypes = 5;
             MaxMergeTypes = 5;
+            NestedModulesLimit = 4;
```

Comment:
Awesome, thanks for the detailed explanation. Logically it makes sense to me, though Dino has the  most familiarity on this and what the edge cases might be, so it'll be great if he can chime in here too.